File: packages/rook_sdk_apple_health/lib/src/data/mapper/data_source_type_mappers.dart
import 'package:rook_sdk_apple_health/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk_core/rook_sdk_core.dart';

extension DataSourceTypeMappers on DataSourceType {
  DataSourceTypeProto toProto() {
    return switch (this) {
      DataSourceType.garmin => DataSourceTypeProto.GARMIN,
      DataSourceType.oura => DataSourceTypeProto.OURA,
      DataSourceType.polar => DataSourceTypeProto.POLAR,
      DataSourceType.fitbit => DataSourceTypeProto.FITBIT,
      DataSourceType.withings => DataSourceTypeProto.WITHINGS,
      DataSourceType.whoop => DataSourceTypeProto.WHOOP,
    };
  }
}

File: packages/rook_sdk_apple_health/ios/Classes/Mapper/DataSourceTypeMappers.swift
//
//  DataSourceTypeMappers.swift
//  rook_sdk_apple_health
//
//  Created by Daniel Nolasco on 02/10/24.
//

import Foundation
import RookSDK

extension DataSourceTypeProto {
    func toDomain() throws -> DataSourceRevoke {
        return switch self {
        case DataSourceTypeProto.garmin:
            DataSourceRevoke.Garmin
        case DataSourceTypeProto.oura:
            DataSourceRevoke.Oura
        case DataSourceTypeProto.polar:
            DataSourceRevoke.Polar
        case DataSourceTypeProto.fitbit:
            DataSourceRevoke.Fitbit
        case DataSourceTypeProto.withings:
            DataSourceRevoke.Withings
        case DataSourceTypeProto.whoop:
            DataSourceRevoke.Whoop
        case .UNRECOGNIZED:
            throw RookSdkPluginErrors.UnknownDataSourceType
        }
    }
}
